### DISCUSSION TITLE
Directus oAuth Provider

### DISCUSSION THREAD

**Discussion by wc-matteo** [directus/directus#6617] [Archive]
I was thinking of having a directus instance dedicated to handle user accounts. Would it be feasible to use this directus as an oAuth provider for other directus instances so users could be stored in a central location and authenticate on these other directus instances with a single account?

@rijkvanzanten @benhaynes

**Comment by benhaynes** [comment]
We've been discussing if/how we might support multiple datastores in Directus 10+. Similar to how we treat Storage Adapters, we could have multiple datastore adapters (SQL, Firebase, etc... maybe even Mongo!). If this is the direction we go, then you would be able to have one datastore manage all system tables, and then manage content in others. I think that would solve this... correct?

**Reply by wc-matteo** [reply]
that sounds really cool! And yeah, it would solve this.
But in the meantime, do you think it would be hard to have a directus instance become an oauth provider?

**Reply by benhaynes** [reply]
Not sure how difficult that would be to create... @rijkvanzanten ? Seems like a nice option to have, especially for Cloud.

**Reply by rijkvanzanten** [reply]
That's a very interesting angle that we haven't explored before. I think that supporting the flow itself wouldn't be too crazy, but we'll have to look into how to configure allowed redirect URLs, and what sort of scopes to support 🤔

**Comment by hostmaster-wikisoft** [comment]
Using a directus instance as a fleksible single source of truth for user/meta data and profiles for a microservice (service as multiple end user app types) system would be awesome. That way companies could build a mini personalized edition of fusion auth for both internal and/or external use and keep it as simple or complex as they wish.
